warehouse/integrations/github/utils.py

```diff
@@ -273,57 +273,61 @@ def _check_signature(self, payload, public_key, signature):
             ) from exc
 
 
-class TokenLeakAnalyzer:
-    def __init__(self, request):
-        self._request = request
-        self._metrics = self._request.find_service(IMetricsService, context=None)
-        self._macaroon_service = self._request.find_service(
-            IMacaroonService, context=None
+def _analyze_disclosure(request, disclosure_record, origin):
```

Comment:
Honnestly, I'd prefer if this method recieved the services it needs instead of the request, but `send_token_compromised_email_leak` requires passing a request. That leaves us with 3 choices:

- Pass the request and use find_service in the function, what I did but not a huge fan
- Pass the services AND the request. Seems unnatural to pass `x` and `f(x)` as arguments to a function. 
- Only pass the services: but then how to send the email ?

Would be easier if email sending used a service too.